pkgs/development/python-modules/ipfshttpclient/default.nix

```
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, py-multiaddr
+, requests
+, pytestCheckHook
+, pytest-cov
+, pytest-dependency
+, pytest-localserver
+, pytest-mock
+, pytest-ordering
+, pytest-cid
+, mock
+, ipfs
+, httpx
+, httpcore
+}:
+
+buildPythonPackage rec {
+  pname = "ipfshttpclient";
+  version = "0.6.1";
+  format = "flit";
+  disabled = pythonOlder "3.5";
+
+  src = fetchFromGitHub {
+    owner = "ipfs-shipyard";
+    repo = "py-ipfs-http-client";
+    rev = version;
+    sha256 = "sha256-wqdI4auGfqXkbTGTZBvrXuAeVN2ZED2sCumFK3mgXSs=";
+  };
+
+  propagatedBuildInputs = [
+    py-multiaddr
+    requests
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+    pytest-cov
+    pytest-dependency
+    pytest-localserver
+    pytest-mock
+    pytest-ordering
+    pytest-cid
+    mock
+    ipfs
+    httpcore
+    httpx
+  ];
+
+  # Remove patch when the package supports the latest IPFS version by default
+  postPatch = ''
+    substituteInPlace ipfshttpclient/client/__init__.py \
+      --replace 'VERSION_MAXIMUM   = "0.7.0"' \
+                'VERSION_MAXIMUM   = "0.9.0"'
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    python3 -X utf8 test/run-tests.py
```

```suggestion
    ${python.interpreter} -X utf8 test/run-tests.py
```

Otherwise it might break with pypy or python2 (theoretically).